### PR TITLE
front end access tokens

### PR DESCRIPTION
**Pull request by mbodeantor**
#### Fixes

* Replace JWT with automatically generated access tokens as a pathway for future RBAC

#### Description

* I created a new API endpoint as an entry point to the quick search endpoint. The new endpoint automatically creates an access token which is stored in the new access_tokens table in Supabase and expires after 5 minutes and isn't visible in the browser which should prevent potential abuse of other endpoints. Let me know if this doesn't alleviate all concerns, but I believe this should be good for launch. We can expand on this in the future if we want to make users login for the quick search app and/or add role based access controls (RBAC) to enable certain levels of access. 

#### Docs

* Gitbook documentation: https://app.gitbook.com/o/-MXypK5ySzExtEzQU6se/s/-MXyolqTg_voOhFyAcr-/~/changes/444/api/endpoints/data-sources-database#undefined

### CODE REVIEW

**Comment by josh-chamberlain** [comment]
nice, so this endpoint would primarily be for the front end? I think we should establish some guidance in the docs for how to decide which endpoint to use.

**Comment by mbodeantor** [comment]
I wondered about that too, I don't think either this or the quick-search endpoint really make sense to use except on the front end, I think we could provide better access to developers through our other endpoints. Maybe it makes sense just to remove them from GitBook and just have them documented in Github?

**Comment by josh-chamberlain** [comment]
@mbodeantor I think keeping them in the same place as fine, while making sure people know what they're for.

**Comment by josh-chamberlain** [comment]
this is fine from my perspective but the other two should weigh in 👀

File: middleware/security.py
Comment (kalenluciano) [inline middleware/security.py:42]: If we're removing JWT here, we need to account for it in the login function in User.py. Right now, we generate an API key but encode it with JWT first before sending it to the frontend. I think we can just create the API key and store it for now. At the very least, we should cut the lines that encode it with JWT and send this false token to the frontend that we can't handle anymore

**Review by kalenluciano — CHANGES_REQUESTED** [review]
This looks like a pretty good solution! I made a few notes of things to change or to account for elsewhere in our codebase. Let me know if you have any questions.

**Comment by mbodeantor** [comment]
@kalenluciano made those changes

File: app.py
Comment (kalenluciano) [inline app.py:25]: I think this only works for me when I do the quick-search route with two arguments. If I do the data-sources search with only one argument, it says that the route isn't found. I'm thinking you can remove <arg1> and <arg2> here and do something like this in the SearchTokens.py file:
`params = request.args.to_dict()` and then access arg1 and arg2 with params

Comment (mbodeantor) [reply app.py:25]: @kalenluciano yeah support for that isn't on this branch. On main, the workaround I used is you have to pass "all" for the arg you want to leave blank

**Review by joshuagraber — COMMENTED** [review]
Looks good. I haven't had a chance to run the app with the changes yet, but the code looks good to me.

**Review by josh-chamberlain — DISMISSED** [review]

**Comment by mbodeantor** [comment]
@kalenluciano I think you have to approve the changes you requestsed

**Review by kalenluciano — DISMISSED** [review]

**Comment by mbodeantor** [comment]
@kalenluciano @josh-chamberlain just need one more approval after I fixed the merge conflict

**Review by josh-chamberlain — APPROVED** [review]

**Pull request merged**
